main/integration-tests/__test__/api-tests/users/create-user.test.js

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+describe('Create user scenarios', () => {
+  let setup;
+  let defaultUser;
+  let username;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    username = await setup.gen.username();
+    const password = await setup.gen.password();
+
+    defaultUser = {
+      username,
+      email: username,
+      password,
+      isAdmin: false,
+      userRole: 'researcher',
+    };
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('create user', () => {
+    it('should fail for anonymous user', async () => {
+      const anonymousSession = await setup.createAnonymousSession();
+      await expect(anonymousSession.resources.users.create(defaultUser)).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
+    });
+
+    it('should fail for inactive admin', async () => {
+      const adminSession = await setup.createAdminSession();
+      await adminSession.resources.currentUser.update({ status: 'inactive', rev: 0 });
+      await expect(adminSession.resources.users.create(defaultUser)).rejects.toEqual(
+        expect.objectContaining({ code: errorCode.http.code.unauthorized }),
+      );
+    });
```

Comment:
I noticed that sometime you use rejects.toMatchObject()  and sometime you use rejects.toEqual(expect.objectContaining())

Is it possible to use rejects.toMatchObject() since this is what most other tests are doing?